### DISCUSSION TITLE
Can you version your supabase backend?

### DISCUSSION THREAD

**Discussion by julienandco** [supabase/supabase#21922] [Questions] [answered]
Hey there,
I've just decided to build my next Flutter project with supabase and I was wondering whether there is a way to version the backend. In my past projects, I would structure the API such as foo-bar-api/v1/do-stuff, foo-bar-api/v2/do-stuff, etc. to be able to introduce breaking changes w/o breaking the apps that are already rolled out and installed on customer devices. Is there any way to do so with supabase? As of now, if i were to change a table column say from int to string, I would get parsing errors in the app, which is why versioning would be nice. The only way I have come up with now is to not use the flutter sdk to access the tables, but instead to only use functions. Even for simple requests as `select * from table-name`. This takes away a lot of development speed though. any thoughts?

**Comment by dshukertjr** [comment, marked as answer]
Here is a great instruction on how you can version your Supabase APIs. 
https://github.com/orgs/supabase/discussions/2957#discussioncomment-1240770
